public/app/features/dashboard-scene/settings/VariablesEditView.tsx

```diff
@@ -168,6 +174,29 @@ export class VariablesEditView extends SceneObjectBase<VariablesEditViewState> i
   public onGoBack = () => {
     this.setState({ editIndex: undefined });
   };
+
+  public onValidateVariableName = (name: string, key: string | undefined): [true, string] | [false, null] => {
+    let errorText = null;
+    if (!RESERVED_GLOBAL_VARIABLE_NAME_REGEX.test(name)) {
+      errorText = "Template names cannot begin with '__', that's reserved for Grafana's global variables";
+    }
+
+    if (!WORD_CHARACTERS_REGEX.test(name)) {
+      errorText = 'Only word characters are allowed in variable names';
+    }
+
+    const variable = this.getVariableSet().getByName(name)?.state;
+
+    if (variable && variable.key !== key) {
+      errorText = 'Variable with the same name already exists';
+    }
+
+    if (errorText) {
+      return [true, errorText];
+    }
+
+    return [false, null];
+  };
 }
```

Comment:
This is easily testable since it is a scene object, what do you think about adding some tests where we ensure we're validating properly the form?